### PR TITLE
create port, probe enhancement

### PR DESCRIPTION
**Pull request by ring630**
1, property terminal -> method get_terminal(create_new_terminal). This is to improve usability before people start to use terminal property.

2, bug fix. PointTerminal.location

3, Refactor. boundary_type, terminal_type.

4, versatile methods to create ports, probes, sources between terminals

### CODE REVIEW

**Comment by ring630** [comment]
> @ring630 @svandenb-dev @SMoraisAnsys Could we merge this PR?

@Samuelopez-ansys I don't have anything to add, so I will merge it once it is reviewed and approved.

File: pyaedt/edb.py
Comment (PipKat) [inline pyaedt/edb.py:3719]: Add "Returns" info.

File: pyaedt/edb.py
Comment (PipKat) [inline pyaedt/edb.py:3749]: Add "Returns" info.

File: pyaedt/edb.py
Comment (PipKat) [inline pyaedt/edb.py:3774]: Add "Returns" docstring section?

File: pyaedt/edb_core/edb_data/padstacks_data.py
Comment (PipKat) [inline pyaedt/edb_core/edb_data/padstacks_data.py:1003]:
```suggestion
        """Get padstack terminal."""
```
Add "Parameters" and "Returns" docstring sections.

File: pyaedt/edb_core/edb_data/terminals.py
Comment (PipKat) [inline pyaedt/edb_core/edb_data/terminals.py:48]: Add a docstring for this property.

File: pyaedt/edb_core/edb_data/terminals.py
Comment (PipKat) [inline pyaedt/edb_core/edb_data/terminals.py:166]:
```suggestion
        """Boundary type.
```

File: pyaedt/edb.py
Comment (svandenb-dev) [inline pyaedt/edb.py:3671]: I think it would be better to move this to edb_data/source.py. I understand it was already here but this bring edb.py a bit messy.

File: pyaedt/edb.py
Comment (svandenb-dev) [inline pyaedt/edb.py:3701]: Same I would move to edb_data/source.py

File: pyaedt/edb.py
Comment (svandenb-dev) [inline pyaedt/edb.py:3731]: Move to edb_data/source.py

File: pyaedt/edb.py
Comment (svandenb-dev) [inline pyaedt/edb.py:3761]: This method should not be located in edb.py as it's too low level

**Review by svandenb-dev — CHANGES_REQUESTED** [review]

File: pyaedt/edb.py
Comment (ring630) [reply pyaedt/edb.py:3761]: @svandenb-dev which class is better to hold this method?

File: pyaedt/edb.py
Comment (svandenb-dev) [reply pyaedt/edb.py:3701]: ok we will postponed this refactoring when pyedb is ready

**Review by svandenb-dev — APPROVED** [review]

**Pull request merged**
